src/main/java/com/pnu/dev/radioserviceapi/service/VideoServiceImpl.java

```diff
@@ -1,7 +1,165 @@
 package com.pnu.dev.radioserviceapi.service;
 
+import com.pnu.dev.radioserviceapi.client.YoutubeApiClient;
+import com.pnu.dev.radioserviceapi.client.dto.YoutubeApiResult;
+import com.pnu.dev.radioserviceapi.client.dto.videos.ItemYoutubeVideosResponse;
+import com.pnu.dev.radioserviceapi.exception.RadioServiceAdminException;
+import com.pnu.dev.radioserviceapi.mongo.Video;
+import com.pnu.dev.radioserviceapi.repository.VideoRepository;
+import com.pnu.dev.radioserviceapi.util.mapper.VideoMapper;
+import com.pnu.dev.radioserviceapi.util.validation.YoutubeVideoIdExtractor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class VideoServiceImpl implements VideoService {
+
+    private final VideoRepository videoRepository;
+
+    private final YoutubeApiClient youtubeApiClient;
+
+    private final VideoMapper videoMapper;
+
+
+    @Autowired
+    public VideoServiceImpl(VideoRepository videoRepository, YoutubeApiClient youtubeApiClient, VideoMapper videoMapper) {
+        this.videoRepository = videoRepository;
+        this.youtubeApiClient = youtubeApiClient;
+        this.videoMapper = videoMapper;
+    }
+
+    @Override
+    public List<Video> findAll() {
+
+        return videoRepository.findAll(Sort.by("priority").ascending());
+    }
+
+    @Override
+    @Transactional
+    public void deleteById(String id) {
+
+        Video video = findVideoByIdOrThrowException(id);
+
+        List<Video> videosToDecrementPriority = videoRepository
+                .findAllByPriorityGreaterThan(video.getPriority());
+
+        videoRepository.deleteById(id);
+
+        decrementPriority(videosToDecrementPriority);
+    }
+
+
+    @Override
+    @Transactional
+    public void create(String link) {
+
+        Video videoFromYoutube = findVideoOnYoutube(link);
+
+        Optional<Video> videoFromDb = videoRepository.findById(videoFromYoutube.getId());
+
+        //Move video to the top if already exists in db
+        if (videoFromDb.isPresent()) {
+            updatePriority(videoFromDb.get(), 1);
+        } else {
+
+            videoFromYoutube = videoFromYoutube
+                    .toBuilder()
+                    .priority(1)
+                    .build();
+
+            List<Video> videosToIncrementPriority = videoRepository
+                    .findAll();
+
+            videoRepository.save(videoFromYoutube);
+            incrementPriority(videosToIncrementPriority);
+        }
+    }
+
+    private Video findVideoOnYoutube(String link) {
+
+        Optional<String> optionalId = YoutubeVideoIdExtractor.getVideoIdFromLink(link);
```

Comment:
String id = YoutubeVideoIdExtractor.getVideoIdFromLink(link).orElseThrow(...);

And `YoutubeVideoIdExtractor` should be a `@Component`